game.cpp

```diff
@@ -1259,48 +1286,54 @@ void test_game() //!OCLINT tests may be many
   }
 #endif
 
+#define FIX_ISSUE_254
 #ifdef FIX_ISSUE_254
   {
     game g;
     put_player_on_food(g.get_player(0), g.get_food()[0]);
     g.tick();
-    assert(is_eaten(g.get_food()[0]))
+    assert(g.get_food()[0].is_eaten());
   }
 #endif
 
+  #define FIX_ISSUE_340
+  #ifdef FIX_ISSUE_340
   // make sure that eat_nth_food() throws a logic_error when the food is already eaten
   {
     game g; //by default one uneaten food
     assert(has_food(g));
-    eat_nth_food(g,0);
+    g.eat_nth_food(0);
     assert(!has_food(g));
     try {
-      eat_nth_food(g,0); // throws exception
+      g.eat_nth_food(0); // throws exception
     }
     catch ( const std::exception& e ) {
       assert(std::string(e.what()) == std::string("You cannot eat food that already has been eaten!"));
     }
   }
+  #endif // FIX_ISSUE_340
+
   // number of food item stays the same,
   // only the state of food item changes after they are eaten
   // eaten food items are ?probably removed by game::tick
   {
     game g; //by default one uneaten food
     const int n_food_items_begin = count_food_items(g);
     assert(has_food(g));
-    eat_nth_food(g,0);
+    g.eat_nth_food(0);
     assert(!has_food(g));
     assert(n_food_items_begin == count_food_items(g));
   }
 
+#define FIX_ISSUE_256
 #ifdef FIX_ISSUE_256
   {
     food f;
     player p;
     put_player_on_food(p, f);
-    assert(player_and_food_are_colliding(p,f));
-    eat_nth_food(g,0);
-    assert(!player_and_food_are_colliding(p,f));
+    assert(in_contact_with_uneaten_food(p,f));
+    f.set_food_state(food_state::eaten);
```

Comment:
Maybe wrap this in a free fucntion
```c++
gets_eaten(food& f)
```
just for the eyes.

Reply:
That would make a great `first` issue! I'm calling dips on writing it 😛

Reply:
Actually, this is exactly the job of `game::eat_food(f)`, although this is not a free function. I'll address this again after tackling the point above, when I get access to Qt again :)

Reply:
Ok, now I think we don't want this `gets_eaten()` function, as this would just be a free copy of `game::eat_food()`, and we would call it for just that one line